be/src/vec/functions/function_string.h

```diff
@@ -386,6 +387,180 @@ class FunctionStrcmp : public IFunction {
+
+                res_data[res_offset[i - 1]] = 'p';
+
+                if (std::strncmp(range_type, "day", 3) == 0 && range_type_size == 3) {
+                    for (int j = i * 3; j < i * 3 + 3; j++) {
+                        memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                               str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                } else if (std::strncmp(range_type, "month", 5) == 0 && range_type_size == 5) {
+                    for (int j = i * 3; j < i * 3 + 2; j++) {
+                        memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                               str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, "01", 2);
+                    curr_len += 2;
+                } else if (std::strncmp(range_type, "year", 4) == 0 && range_type_size == 4) {
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[i * 3].c_str(),
+                           str_v[i * 3].size());
+                    curr_len += str_v[i * 3].size();
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, "0101", 4);
+                    curr_len += 4;
+                } else {
+                    return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                            "The first argument is range, the second argument muse be "
+                            "day, month or year ");
+                }
+                memcpy(&res_data[res_offset[i - 1]] + curr_len, "000000", 6);
+                curr_len += 6;
+                res_offset[i] = res_offset[i - 1] + curr_len;
+            }
+
+            block.get_by_position(result).column = std::move(res);
+            return Status::OK();
+        } else {
+            return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                    "Partition type must be range or list");
+        }
+
+        return Status::OK();
+    }
+
+private:
+    std::u16string _string_to_u16string(const std::string& str) const {
+        std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> convert;
+        return convert.from_bytes(str);
+    }
+
+    std::string _string_to_unicode(const std::u16string& s) const {
+        std::string res_s;
+        if (s.length() > 0 && s[0] == '-') {
+            res_s += '_';
+        }
+        for (int i = 0; i < s.length(); i++) {
+            char ch = s[i];
+            if (std::isalnum(ch)) {
+                res_s += ch;
+            } else {
+                int unicodeValue = _get_code_foint_at(s, i);
+                char buffer[18];
+                sprintf(buffer, "%x", unicodeValue);
+                res_s += buffer;
+            }
+        }
+        return res_s;
+    }
+
+    int _get_code_foint_at(const std::u16string& str, std::size_t index) const {
+        char16_t first = str[index];
```

Comment:
after changed `_string_to_unicode`, use `index_check_const` to deal with `index`